src/language.h

```diff
@@ -168,14 +162,21 @@ class Language {
 	void Clear();
 
 	uint16 RegisterStrings(const TextData &td, const char * const names[], uint16 base = STR_GENERIC_END);
+	void InitMetaInfo();
 
-	std::string GetText(StringID number);
+	std::string GetSgText(StringID number);
+	std::string GetPlural(StringID number, int64 amount);
 	std::string GetLanguageName(int lang_index);
+	uint GetPluralFormIndex(int lang_index, int64 amount);
 
 private:
 	/** Registered strings. Entries may be \c nullptr for unregistered or non-existing strings. */
 	const TextString *registered[2048]; // Arbitrary size.
 	uint first_free; ///< 'First' string index that is not allocated yet.
+
+	std::unique_ptr<EvaluateableExpression> plural_forms[LANGUAGE_COUNT];  ///< Compiled expression for every language's plural rule.
+	int nplurals                                        [LANGUAGE_COUNT];  ///< Number of plural forms in each language.
+	int singular_form_index                             [LANGUAGE_COUNT];  ///< The plural form index for amount 1, cached for performance.
```

Comment:
Compiled expression fit better in a language class as method. Same with `nplurals`.
Give the compiler a chance to understand these things are used together.

Singular form shouldn't be special enough to warrant caching, I think.